src/ILLink.Tasks/CreateRuntimeRootDescriptorFile.cs

```diff
@@ -139,11 +169,33 @@ void ProcessMscorlib (string typeFile)
 			string[] types = File.ReadAllLines (typeFile);
 			DefineTracker defineTracker = new DefineTracker (defineConstants, Log, typeFile);
 			string classId;
+			FeatureSwitchMembers? currentFeatureSwitch = null;
 
 			foreach (string def in types) {
 				if (defineTracker.ProcessLine (def) || !defineTracker.IsActiveSection)
 					continue;
 
+				//We will handle BEGIN_ILLINK_FEATURE_SWITCH and END_ILLINK_FEATURE_SWITCH
+				if (def.StartsWith ("BEGIN_ILLINK_FEATURE_SWITCH")) {
+					if (currentFeatureSwitch.HasValue) {
+						Log.LogError ($"Could not figure out feature switch status in '{typeFile}' for line {def}");
+					} else {
+						char[] separators = { ',', '(', ')', ' ', '\t', '/' };
+						string[] featureSwitchElements = def.Split (separators, StringSplitOptions.RemoveEmptyEntries);
+						currentFeatureSwitch = new FeatureSwitchMembers (featureSwitchElements[1], featureSwitchElements[2], featureSwitchElements[3]);
```

Comment:
we should check that the returned number of segments is as expected and fail otherwise... to avoid surprises.